src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs

```diff
@@ -118,7 +117,68 @@ private static void OnAutoPlayChanged(DependencyObject sender, DependencyPropert
 
 		#endregion
 
-		#region MediaPlayer Property
+		#region IsFullWindow Property
+
+		public bool IsFullWindow
+		{
+			get { return (bool)GetValue(IsFullWindowProperty); }
+			set { SetValue(IsFullWindowProperty, value); }
+		}
+
+		public static DependencyProperty IsFullWindowProperty { get; } =
+			DependencyProperty.Register(
+				nameof(IsFullWindow),
+				typeof(bool),
+				typeof(MediaPlayerElement),
+				new FrameworkPropertyMetadata(false, OnIsFullWindowChanged));
+
+
+		private static void OnIsFullWindowChanged(DependencyObject sender, DependencyPropertyChangedEventArgs args)
+		{
+			sender.Maybe<MediaPlayerElement>(mpe =>
+			{
+				mpe.ToogleFullScreen((bool)args.NewValue);
+			});
+		}
+
+		private object _mainContent;
+
+		private void ToogleFullScreen(bool showFullScreen)
+		{
+			if (showFullScreen)
+			{
+				Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
+				{
+					ApplicationView.GetForCurrentView().TryEnterFullScreenMode();
+				});
+
+				_mainContent = (Windows.UI.Xaml.Window.Current.Content as Frame).Content;
+				(Windows.UI.Xaml.Window.Current.Content as Frame).Content = _layoutRoot;
+
+#if __ANDROID__
+				this.RemoveView(_layoutRoot);
```

Comment:
Shouldn't you call this first? I'm surprised it works.

Reply:
It works. But I'll check that, it's maybe clearer

Reply:
Fixed

Reply:
Done